modules/images/regenerate-existing-images/background-process/class-perflab-background-job.php

```diff
@@ -0,0 +1,403 @@
+	}
+
+	/**
+	 * Creates a new job in the queue.
+	 *
+	 * Job refers to the term and queue refers to the `background_job` taxonomy.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $name Name of job identifier.
+	 * @param array  $data Data for the job.
+	 *
+	 * @return Perflab_Background_Job|WP_Error Job object if created successfully, else WP_Error.
+	 */
+	public static function create( $name, array $data = array() ) {
+		// Insert the new job in queue.
+		$term_data = wp_insert_term( 'job_' . time(), PERFLAB_BACKGROUND_JOB_TAXONOMY_SLUG );
+
+		if ( ! is_wp_error( $term_data ) ) {
+			update_term_meta( $term_data['term_id'], self::JOB_DATA_META_KEY, $data );
+			update_term_meta( $term_data['term_id'], self::JOB_NAME_META_KEY, $name );
+
+			/**
+			 * Create a fresh instance to return.
+			 *
+			 * @var Perflab_Background_Job
+			 */
+			$job = new self( $term_data['term_id'] );
+
+			$job->name = $name;
+			$job->data = $data;
+
+			// Set the queued status for freshly created jobs.
+			$job->set_status( self::JOB_STATUS_QUEUED );
+
+			return $job;
+		}
+
+		return $term_data;
+	}
+
+	/**
+	 * Returns the number of attempts executed for a job.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return int
+	 */
+	public function get_attempts() {
+		$attempts = get_term_meta( $this->job_id, self::JOB_ATTEMPTS_META_KEY, true );
+
+		return absint( $attempts );
+	}
+
+	/**
+	 * Set the start time of job.
+	 * It tells at what point of time the job has been started.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param int $time Timestamp (in seconds) at which job has been started.
+	 *
+	 * @return void
+	 */
+	public function lock( $time = null ) {
+		$time = empty( $time ) ? time() : $time;
+		update_term_meta( $this->job_id, self::JOB_LOCK_META_KEY, $time );
+		$this->set_status( self::JOB_STATUS_RUNNING );
+	}
+
+	/**
+	 * Get the timestamp (in seconds) when the job was started.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return int Start time of the job.
+	 */
+	public function get_start_time() {
+		$time = get_term_meta( $this->job_id, self::JOB_LOCK_META_KEY, true );
+
+		return absint( $time );
+	}
+
+	/**
+	 * Unlocks the process.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return void
+	 */
+	public function unlock() {
+		delete_term_meta( $this->job_id, self::JOB_LOCK_META_KEY );
+	}
+
+	/**
+	 * Checks if the background job is completed.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return bool
+	 */
+	public function is_completed() {
+		return ( self::JOB_STATUS_COMPLETE === $this->get_status() );
+	}
+
+	/**
+	 * Checks that the job term exist.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array|null
+	 */
+	private function exists() {
+		return term_exists( $this->job_id, PERFLAB_BACKGROUND_JOB_TAXONOMY_SLUG );
+	}
+
+	/**
+	 * Checks if the job is running.
+	 *
+	 * If the job lock is present, it means job is running.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return bool
+	 */
+	private function is_running() {
+		return ( self::JOB_STATUS_RUNNING === $this->get_status() );
+	}
```

Comment:
Just curious: Why is `is_completed()` public but `is_running()` private? They do kind of the same thing (convenience wrappers for comparing the `get_status()` result to one of the allowed values), so I would assume we should align the access level here.

Reply:
Made both public. `is_running` was only being used inside class, but `is_completed` was being used by while loop in process runner class, so it was made public.

I agree with your point that both the methods should be public as those are just utility methods/wrappers.